src/tracer/tracer.ts

```diff
@@ -225,20 +234,87 @@ export class AutoblocksTracer {
     });
   }
 
-  public async sendEvent(message: string, args?: SendEventArgs): Promise<void> {
-    try {
-      // This store should only be set in the context of a test run
-      const store = testCaseRunAsyncLocalStorage.getStore();
-      if (store) {
-        await this.sendTestEventUnsafe(message, args);
-      } else {
-        await this.sendEventUnsafe(message, args);
+  private sendEventUnsafe(message: string, args?: SendEventArgs): void {
+    // Generate the timestamp now so that it reflects the time
+    // the event was sent and not when the task was picked up
+    // by the event loop
+    const argsWithTimestamp: SendEventArgs = {
+      timestamp: makeISOTimestamp(),
+      // Our timestamp will be overridden with the user's timestamp
+      // if it's set
+      ...args,
+    };
+
+    const task: BackgroundTask = {};
+    backgroundTasks.add(task);
```

Comment:
How does this work? Wouldn't the set only ever have a length of 1 since you are inserting `{}` each time?

Reply:
![image](https://github.com/autoblocksai/javascript-sdk/assets/15793238/c378a6be-d5c6-441e-bcd9-478fff849785)
objects are by reference, not value

Reply:
ya what adam said, but it does look weird. i'll add a comment